src/transformers/data/data_collator.py

```diff
@@ -34,30 +34,50 @@ def default_data_collator(features: List[InputDataClass]) -> Dict[str, torch.Ten
     # So we will look at the first element as a proxy for what attributes exist
     # on the whole batch.
     first = features[0]
+    batch = {}
+
+    # Note: below there are a few mysterious torch.stack([torch.tensor(f) for f ...]).
```

Comment:
This things happens four times below, I put `# See comment in the intro` notes to document it but maybe the corresponding code deserves its own function with the code documented there? Let me know what you prefer.